digitalocean/datasource_digitalocean_database_replica_test.go

```diff
@@ -0,0 +1,82 @@
+	var databaseReplica godo.DatabaseReplica
+	var database godo.Database
+
+	databaseName := randomTestName()
+	databaseReplicaName := randomTestName()
+
+	databaseConfig := fmt.Sprintf(testAccCheckDigitalOceanDatabaseClusterConfigBasic, databaseName)
+	replicaConfig := fmt.Sprintf(testAccCheckDigitalOceanDatabaseReplicaConfigBasic, databaseReplicaName)
+	datasourceReplicaConfig := fmt.Sprintf(testAccCheckDigitalOceanDatasourceDatabaseReplicaConfigBasic, "digitalocean_database_replica.read-01.cluster_id", databaseReplicaName)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviderFactories,
+		CheckDestroy:      testAccCheckDigitalOceanDatabaseReplicaDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: databaseConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDigitalOceanDatabaseClusterExists("digitalocean_database_cluster.foobar", &database),
+				),
+			},
+			{
+				Config: databaseConfig + replicaConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDigitalOceanDatabaseReplicaExists("digitalocean_database_replica.read-01", &databaseReplica),
+					testAccCheckDigitalOceanDatabaseReplicaAttributes(&databaseReplica, databaseReplicaName),
+					resource.TestCheckResourceAttr(
+						"digitalocean_database_replica.read-01", "size", "db-s-2vcpu-4gb"),
+					resource.TestCheckResourceAttr(
+						"digitalocean_database_replica.read-01", "region", "nyc3"),
+					resource.TestCheckResourceAttr(
+						"digitalocean_database_replica.read-01", "name", databaseReplicaName),
+					resource.TestCheckResourceAttrSet(
+						"digitalocean_database_replica.read-01", "host"),
+					resource.TestCheckResourceAttrSet(
+						"digitalocean_database_replica.read-01", "private_host"),
+					resource.TestCheckResourceAttrSet(
+						"digitalocean_database_replica.read-01", "port"),
+					resource.TestCheckResourceAttrSet(
+						"digitalocean_database_replica.read-01", "user"),
+					resource.TestCheckResourceAttrSet(
+						"digitalocean_database_replica.read-01", "uri"),
+					resource.TestCheckResourceAttrSet(
+						"digitalocean_database_replica.read-01", "private_uri"),
+					resource.TestCheckResourceAttrSet(
+						"digitalocean_database_replica.read-01", "password"),
+					resource.TestCheckResourceAttr(
+						"digitalocean_database_replica.read-01", "tags.#", "1"),
+					resource.TestCheckResourceAttrSet(
+						"digitalocean_database_replica.read-01", "private_network_uuid"),
+				),
+			},
+			{
+				Config: databaseConfig + replicaConfig + datasourceReplicaConfig,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrPair("digitalocean_database_replica.read-01", "cluster_id",
+						"digitalocean_database_replica.my_db_replica", "cluster_id"),
+					resource.TestCheckResourceAttrPair("digitalocean_database_replica.read-01", "name",
+						"digitalocean_database_replica.my_db_replica", "name"),
+				),
```

Comment:
```suggestion
					resource.TestCheckResourceAttrPair("digitalocean_database_replica.read-01", "host",
						"data.digitalocean_database_replica.my_db_replica", "host"),
								),
```

Would be great to test that the computed attributes are set as expected. In fact, it's more important to test that here than in the step above which is really just setting things up for this test.

Reply:
We were troubleshooting the id issue but this was next